Zend/zend_language_parser.y

```diff
@@ -548,6 +548,10 @@ function_declaration_statement:
 	backup_fn_flags '{' inner_statement_list '}' backup_fn_flags
 		{ $$ = zend_ast_create_decl(ZEND_AST_FUNC_DECL, $2 | $13, $1, $4,
 		      zend_ast_get_str($3), $6, NULL, $11, $8, NULL); CG(extra_fn_flags) = $9; }
+   | function returns_ref T_STRING backup_doc_comment '(' parameter_list ')' return_type
+        backup_fn_flags T_DOUBLE_ARROW expr ';' backup_fn_flags
+     		{ $$ = zend_ast_create_decl(ZEND_AST_FUNC_DECL, $2, $1, $4,
```

Comment:
easy to fix: Should this be `$2 | $13` instead like the above declaration? The last occurrence of backup_fn_flags is $13, and that includes flags set while parsing the `expr`

Can you add a test of the expected behavior of `return yield $param` for `stdClass $param`? 


```
	|	T_YIELD { $$ = zend_ast_create(ZEND_AST_YIELD, NULL, NULL); CG(extra_fn_flags) |= ZEND_ACC_GENERATOR; }
	|	T_YIELD expr { $$ = zend_ast_create(ZEND_AST_YIELD, $2, NULL); CG(extra_fn_flags) |= ZEND_ACC_GENERATOR; }
	|	T_YIELD expr T_DOUBLE_ARROW expr { $$ = zend_ast_create(ZEND_AST_YIELD, $4, $2); CG(extra_fn_flags) |= ZEND_ACC_GENERATOR; }
	|	T_YIELD_FROM expr { $$ = zend_ast_create(ZEND_AST_YIELD_FROM, $2); CG(extra_fn_flags) |= ZEND_ACC_GENERATOR; }
```

Reply:
I hadn't even though to test generators with this syntax, good call.

I tried it, and `function read($it): iterable => yield from $it;` does work already, albeit that's not very useful.  `yield from $a` where $a is a `stdClass` gives an error on its own, so that seems pre-existing.  I'm not sure what you mean to test here.

Reply:
> I'm not sure what you mean to test here.

I would have thought there would be some sort of impact of not adding those flags to the ZEND_AST_FUNC_DECL

```
/* {{{ Returns whether this function is a generator */
ZEND_METHOD(ReflectionFunctionAbstract, isGenerator)
{
	_function_check_flag(INTERNAL_FUNCTION_PARAM_PASSTHRU, ZEND_ACC_GENERATOR);
}
/* }}} */
```

E.g. adding a test of ReflectionFunction->isGenerator is one thing I had in mind, or whatever ZEND_ACC_GENERATOR would affect.

Reply:
In case anyone's wondering, `return yield from` does return something useful - if the expression for `yield from` is a generator, it returns Generator->getReturn() - otherwise it returns null. 

With that said, I think use cases would be very rare, but exist.

```
php > function example() { yield 'first'; return 'retval'; }
php > function test() { echo "in test\n"; return yield from example(); }
php > $gen = test();
php > foreach ($gen as $value) { echo "value: $value\n"; }
in test
value: first
php > var_export($gen->getReturn());
'retval'
```

```c
	/* This is the default return value
	 * when the expression is a Generator, it will be overwritten in zend_generator_resume() */
	if (RETURN_VALUE_USED(opline)) {
		ZVAL_NULL(EX_VAR(opline->result.var));
	}
```